promdump/promdump_test.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"io"
+	"log"
+	"testing"
+	"time"
+)
+
+func setupLogging() {
+	// Need this to avoid crashing when a func under test tries to log
+	//logger = log.Default()
+	logger = log.New(io.Discard, "", 0)
+}
+
+func freezeTime() {
+	// The promdump code uses a variable called now instead of using time.Now directly so that we can replace calls to
+	// now() with a custom func for testing purposes.
+	staticNow := time.Now().Round(time.Second).UTC()
+	log.Printf("freezeTime: overriding now() to always return %v for testing purposes", staticNow.Format(time.RFC3339))
+	now = func() time.Time {
+		return staticNow
+	}
+}
+
+func TestGetMetricName(t *testing.T) {
+	setupLogging()
+
+	testMetrics := map[string]*promExport{
+		"emptymetric": {collect: true, changedFromDefault: false, requiresNodePrefix: false},
+		"exporter":    {exportName: "exporter_export", collect: true, changedFromDefault: false, requiresNodePrefix: true},
+		"jobmetric":   {jobName: "jobmetric", collect: true, changedFromDefault: false, requiresNodePrefix: false},
+		"badmetric":   {exportName: "badmetric_export", jobName: "badmetric", collect: true, changedFromDefault: false, requiresNodePrefix: true},
+	}
+
+	testMetric := testMetrics["emptymetric"]
```

Comment:
This code block is essentially repeated 4 times.

I would suggest putting the err message as a part each metric.
This will allow adding tests without creating more (repeated) code blocks.